Sources/Smithy/Logging/LogAgent.swift

```diff
@@ -39,80 +41,85 @@ public enum LogAgentLevel: String, Codable, CaseIterable {
                  line: line)
     }
 
     /// Log a message passing with the `.debug` log level.
-    func debug(_ message: String, file: String = #file, function: String = #function, line: UInt = #line) {
+    func debug(_ message: String, file: String = #fileID, function: String = #function, line: UInt = #line) {
         self.log(level: .debug,
                  message: message,
                  metadata: nil,
-                 source: Self.currentModule(),
+                 source: currentModule(fileID: file),
                  file: file,
                  function: function,
                  line: line)
     }
 
     /// Log a message passing with the `.error` log level.
-    func error(_ message: String, file: String = #file, function: String = #function, line: UInt = #line) {
+    func error(_ message: String, file: String = #fileID, function: String = #function, line: UInt = #line) {
         self.log(level: .error,
                  message: message,
                  metadata: nil,
-                 source: Self.currentModule(),
+                 source: currentModule(fileID: file),
                  file: file,
                  function: function,
                  line: line)
     }
 
     /// Log a message passing with the `.trace` log level.
-    func trace(_ message: String, file: String = #file, function: String = #function, line: UInt = #line) {
+    func trace(_ message: String, file: String = #fileID, function: String = #function, line: UInt = #line) {
         self.log(level: .trace,
                  message: message,
                  metadata: nil,
-                 source: Self.currentModule(),
+                 source: currentModule(fileID: file),
                  file: file,
                  function: function,
                  line: line)
     }
 
     /// Log a message passing with the `.fatal` log level.
-    func fatal(_ message: String, file: String = #file, function: String = #function, line: UInt = #line) {
+    func fatal(_ message: String, file: String = #fileID, function: String = #function, line: UInt = #line) {
         self.log(level: .fatal,
                  message: message,
                  metadata: nil,
-                 source: Self.currentModule(),
+                 source: currentModule(fileID: file),
                  file: file,
                  function: function,
                  line: line)
     }
 }
+
+/// Parses the module name from `#fileID`.
+///
+/// Instructions for parsing module from `#fileID` are here:
+/// https://developer.apple.com/documentation/swift/fileid()
+/// - Parameter fileID: The value of the `#fileID` macro at the point of logging.
+/// - Returns: The name of the module, as parsed from the passed `#fileID`.
+private func currentModule(fileID: String) -> String {
+    let utf8All = fileID.utf8
+    if let slashIndex = utf8All.firstIndex(of: UInt8(ascii: "/")) {
+        return String(fileID[..<slashIndex])
+    } else {
+        return "n/a"
+    }
+}
```

Comment:
Compare this `currentModule()` function to the one deleted above; this one is significantly simpler because `#fileID` always has the module name right out front, followed by a `/`.

Reply:
Also I should point out this is exactly how `swift-log` makes this determination for itself:
https://github.com/apple/swift-log/blob/main/Sources/Logging/Logging.swift#L1523